kratos/tests/cpp_tests/modified_shape_functions/test_tetrahedra_3d_4_ausas_incised_shape_functions.cpp

```diff
@@ -0,0 +1,602 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Franziska Wahl
+//
+
+// Project includes
+#include "testing/testing.h"
+#include "containers/model.h"
+#include "includes/checks.h"
+#include "utilities/divide_tetrahedra_3d_4.h"
+#include "modified_shape_functions/tetrahedra_3d_4_ausas_incised_shape_functions.h"
+
+namespace Kratos
+{
+    namespace Testing
+    {
+
+        KRATOS_TEST_CASE_IN_SUITE(AusasIncisedShapeFunctionsTetrahedra3D4Horizontal, KratosCoreFastSuite)
+        {
+            Model current_model;
+
+            // Generate a model part with the previous
+            ModelPart& base_model_part = current_model.CreateModelPart("Tetrahedra");
+            base_model_part.AddNodalSolutionStepVariable(DISTANCE);
+
+            // Fill the model part geometry data
+            base_model_part.CreateNewNode(1, 0.0, 0.0, 0.0);
+            base_model_part.CreateNewNode(2, 1.0, 0.0, 0.0);
+            base_model_part.CreateNewNode(3, 0.0, 1.0, 0.0);
+            base_model_part.CreateNewNode(4, 0.0, 0.0, 1.0);
+            Properties::Pointer p_properties(new Properties(0));
+            base_model_part.CreateNewElement("Element3D4N", 1, {1, 2, 3, 4}, p_properties);
+
+            // Set the elemental distances vector
+            Geometry<Node<3>>::Pointer p_geometry = base_model_part.Elements()[1].pGetGeometry();
+
+            array_1d<double, 4> distances_vector;
+            distances_vector[0] = -1.0;
+            distances_vector[1] = -1.0;
+            distances_vector[2] = -1.0;
+            distances_vector[3] =  1.0;
+            array_1d<double, 6> edge_dist_extra_vector;
+            edge_dist_extra_vector[0] = -1.0;
+            edge_dist_extra_vector[1] = -1.0;
+            edge_dist_extra_vector[2] = -1.0;
+            edge_dist_extra_vector[3] =  0.5;
+            edge_dist_extra_vector[4] = -1.0;
+            edge_dist_extra_vector[5] =  0.5;
```

Comment:
This doesn't look very nice and I've come accross this a couple of times, but there isn't any direct assignment when declaring an `Kratos::array_1d` available, except for `ZeroVector`, is there?

Reply:
Actually it's better if you use an `std::array` since you can do
```c++
std::array<double,6> values_to_check = {{1.0,2.0,3.0,4.0,5.0,6.0}};
```